File: androidApp/src/main/java/dev/haroldjose/familysharedlist/android/dependencyInjection/nativeModule.kt
package dev.haroldjose.familysharedlist.android.dependencyInjection

import dev.haroldjose.familysharedlist.android.dataLayer.datasource.local.keyValueStorage.KeyValueStorageDataSource
import dev.haroldjose.familysharedlist.android.services.firebase.FirebaseAnalytics
import dev.haroldjose.familysharedlist.dependencyInjection.makeNativeModule

val nativeModule = makeNativeModule(
    firebaseAnalytics = { FirebaseAnalytics() },
    keyValueStorageDataSource = { KeyValueStorageDataSource() }
)

File: iosApp/iosApp/dependencyInjection/koin/nativeModule.swift
import Foundation
import shared

var nativeModule: Koin_coreModule = MakeNativeModuleKt.makeNativeModule(
    firebaseAnalytics: { _ in FirebaseAnalytics() }, 
    firebaseCrashlytics: { _ in FirebaseCrashlytics() },
    keyValueStorageDataSource: { _ in KeyValueStorageDataSource() }
)
